src/main/java/com/aws/greengrass/clientdevices/auth/metrics/MetricsEmitter.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.clientdevices.auth.metrics;
+
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import javax.inject.Inject;
+
+public class MetricsEmitter {
+    private ScheduledFuture<?> future;
+    private final ScheduledExecutorService ses;
+    private final Object emitMetricsLock = new Object();
+    private static final int DEFAULT_PERIODIC_AGGREGATE_INTERVAL_SEC = 3_600;
+    private final ClientDeviceAuthMetrics metrics;
+
+    /**
+     * Constructor for the MetricsEmitter.
+     *
+     * @param ses     {@link ScheduledExecutorService}
+     * @param metrics {@link ClientDeviceAuthMetrics}
+     */
+    @Inject
+    public MetricsEmitter(ScheduledExecutorService ses, ClientDeviceAuthMetrics metrics) {
+        this.ses = ses;
+        this.metrics = metrics;
+    }
+
+    /**
+     * Emit metrics at an interval.
+     */
+    public void emit() {
```

Comment:
What calls this?

Reply:
Your call if we can plumb it through or not. This may not be the interface we want. `start()`/`stop()` maybe?

This will probably be called from ClientDeviceAuthService, yes